pages/api/search/index.js

```diff
@@ -22,39 +22,25 @@ export default async function handler(req, res) {
     .trim()
     .replace(/\s{2,}/g, " ")
     .toLowerCase();
-  const terms = cleanedSlug.split(",");
-  let filteredUsers = [];
 
+  const terms = cleanedSlug.split(",");
   try {
-    const users = await getProfiles();
-
-    filteredUsers = users.flat().filter((user) => {
-      const nameLower = user.name.toLowerCase();
-      const usernameLower = user.username.toLowerCase();
-      const userTagsString = user.tags.join(", ").toLowerCase();
-      const userLocationString = user.location
-        ? user.location.provided.toLowerCase()
-        : "";
-
-      const isUserMatched = terms.every((term) => {
-        const cleanedTerm = term.trim();
-
-        return (
-          usernameLower.includes(cleanedTerm) ||
-          nameLower.includes(cleanedTerm) ||
-          userTagsString.includes(cleanedTerm) ||
-          userLocationString.includes(cleanedTerm)
-        );
-      });
-      return isUserMatched;
-    });
+    const filteredUsers = await searchUsers(terms);
 
     if (!filteredUsers.length) {
       return res.status(404).json({ error: `${cleanedSlug} not found` });
     }
 
     res.status(200).json({ users: filteredUsers });
   } catch (e) {
+    if (e.code === 40324) {
```

Comment:
It's preferable not to hardcode the values in the code; instead, you can move them to constants.
f.e. 
`res.status(OKAY_STATUS_CODE);`
`if (e.code === AUTOCOMPLETE_ERROR_CODE) `

Reply:
I think this is a great suggestion 👍